contents/tutorials/ios-session-replays.md

```diff
@@ -0,0 +1,189 @@
+            Text("Welcome!")
+                .accessibilityIdentifier("ph-no-capture")
+                .font(.largeTitle)
+                .padding()
+            
+            Button(action: {
+                isLoggedIn = false
+            }) {
+                Text("Logout")
+                    .frame(maxWidth: .infinity)
+                    .padding()
+                    .background(Color.red)
+                    .foregroundColor(.white)
+                    .cornerRadius(10)
+            }
+            .padding(.top, 20)
+        }
+        .padding()
+    }
+}
+
+#Preview {
+    ContentView()
+}
+```
+
+Our basic setup is now complete. Build and run your app to see it in action.
+
+![Video of basic iOS app setup](https://res.cloudinary.com/dmukukwp6/video/upload/v1720536097/posthog.com/contents/demo-app-ios.mp4)
+
+## 2. Add PostHog to your app
+
+With our app set up, it’s time to install and set up PostHog. If you don't have a PostHog instance, you can [sign up for free](https://us.posthog.com/signup).
+
+First, add [`posthog-ios`](/docs/libraries/ios) as a dependency to your app using [Swift Package Manager](https://developer.apple.com/documentation/xcode/adding-package-dependencies-to-your-app) (or if you prefer, you can use [CocoaPods](/docs/libraries/ios#cocoapods)).
+
+To add the package dependency to your Xcode project, select `File > Add Package Dependency` and enter the URL `https://github.com/PostHog/posthog-ios.git`. Select `posthog-ios` and click Add Package.
+
+Note that session replays require SDK version `3.6.0` or higher.
+
+![Add PostHog from Swift Package Manager](https://res.cloudinary.com/dmukukwp6/image/upload/v1720532354/posthog.com/contents/Screenshot_2024-07-09_at_2.32.30_PM.png)
+
+Next, configure your PostHog instance in `App.swift` using your project API key and instance address (you can find these in [your project settings](https://us.posthog.com/project/settings)):
+
+```swift file=App.swift
+import SwiftUI
+import PostHog
+
+@main
+struct posthog_session_replayApp: App {
+    init() {
+        let POSTHOG_API_KEY = "<ph_project_api_key>"
+        let POSTHOG_HOST = "<ph_client_api_host>"  // usually 'https://us.i.posthog.com' or 'https://eu.i.posthog.com'
+        let configuration = PostHogConfig(apiKey: POSTHOG_API_KEY, host: POSTHOG_HOST)
+        configuration.sessionReplay = true
+        configuration.sessionReplayConfig.screenshotMode = true // required for SwiftUI session replays
+        PostHogSDK.shared.setup(configuration)
+    }
+    
+    var body: some Scene {
+        WindowGroup {
+            ContentView()
+        }
+    }
+}
+```
+
+To check your setup, build and run your app a few times. Enter in any values in the textfields and click **Log in**. You should start session replays in the [Session replay tab](https://us.posthog.com/replay/recent) in PostHog 🎉
+
+<ProductScreenshot
+  imageLight={""} 
+  imageDark={""} 
+  alt="iOS session replays in PostHog" 
+  classes="rounded"
+/>
```

Comment:
@marandaneto I see this redacted view for all texts and inputs on the screen. Am I missing something?

<img width="1121" alt="Screenshot 2024-07-10 at 12 59 57 PM" src="https://github.com/PostHog/posthog.com/assets/7090054/f8dec4d9-7a9d-4b9d-bce1-85edc71594df">

Reply:
re https://github.com/PostHog/posthog.com/pull/8934/files#r1691519516
did you try disabling the automatic masking?

Reply:
@Lior539 can you also try `configuration.sessionReplayConfig.maskAllImages = false`

Reply:
Some times we don't know if SwiftUI views are an image or a text, most of the time, we redact if either option is enabled for safety, sometimes we do, it depends on how the views are declared and how the views are transformed to native views.

Reply:
in your case, the issue is also this https://github.com/PostHog/posthog-ios/blob/6377be7d8423df126966ddf83d62c34458a8625b/PostHog/Replay/UITextInputTraits%2BUtil.swift#L12-L31
if you define a text with `.keyboardType(.emailAddress)` or `SecureField`, we ,automatically redact because we know they are sensitive for sure, so even if redacting is disabled, in this case, we always redact for safety, so people dont have to do this manually

Reply:
In this case `Button` is an image, but contains a label that can be sensitive, so if either option is enabled, we redact, the only way to get the button not redacted, is if you disable both redacting :( sadly but its a SwiftUI limitation.

Reply:
@marandaneto thanks that make sense. However, the "welcome" text is also redacted on the next screen. This is just a text component. Any ideas why?

```
    var body: some View {
        VStack {
            Text("Welcome!")
                .font(.largeTitle)
                .padding()
```

<img width="961" alt="Screenshot 2024-07-29 at 1 29 56 PM" src="https://github.com/user-attachments/assets/638d9ba5-bfcd-439f-b880-dd8a93f17956">

<img width="451" alt="Screenshot 2024-07-29 at 1 32 00 PM" src="https://github.com/user-attachments/assets/d144e39c-b539-4544-b2af-971da835f5ba">